src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -572,6 +573,76 @@ private static bool HasApplicableConditionalMethod(OverloadResolutionResult<Meth
             return result;
         }
 
+        private BoundExpression BindDynamicArgToLocalFunction(
```

Comment:
The name implies that the function binds an argument, but it actually binds an invocation. Consider renaming to BindLocalFunctionInvocationWithDynamicArgument.